src/SSHDebugPS/OutputWindow.cs

```diff
@@ -0,0 +1,123 @@
+        }
+
+        private IVsOutputWindow outputWindow;
+        private IVsUIShell shell;
+
+        private Dictionary<string, PaneInfo> panes = new Dictionary<string, PaneInfo>()
+        {
+            // The 'Debug' pane exists by default
+            { "Debug", new PaneInfo(VSConstants.GUID_OutWindowDebugPane) }
+        };
+
+        private OutputWindow()
+        {
+            outputWindow = Package.GetGlobalService(typeof(SVsOutputWindow)) as IVsOutputWindow;
+            shell = Package.GetGlobalService(typeof(SVsUIShell)) as IVsUIShell;
+            if (outputWindow == null || shell == null)
+            {
+                throw new NullReferenceException("Could not create OutputWindow");
+            }
+        }
+
+        /// <summary>
+        /// Writes text directly to the VS Output window.
+        /// </summary>
+        public void Write(string message, string pane = "Debug")
+        {
+            ThreadHelper.JoinableTaskFactory.Run(async delegate
+            {
+                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                try
+                {
+                    // Get the pane guid
+                    PaneInfo paneInfo;
+                    if (!panes.TryGetValue(pane, out paneInfo))
+                    {
+                        // Pane didn't exist, create it
+                        paneInfo = new PaneInfo(Guid.NewGuid());
+                        panes.Add(pane, paneInfo);
+                    }
+
+                    // Get the pane
+                    IVsOutputWindowPane outputPane;
+                    if (outputWindow.GetPane(ref paneInfo.paneId, out outputPane) != VSConstants.S_OK)
+                    {
+                        // Failed to get the pane - might need to create it first
+                        outputWindow.CreatePane(ref paneInfo.paneId, pane, fInitVisible: 1, fClearWithSolution: 1);
+                        outputWindow.GetPane(ref paneInfo.paneId, out outputPane);
+                    }
+
+                    // The first time we output text to a pane, ensure it's visible
+                    if (!paneInfo.Shown)
+                    {
+                        paneInfo.Shown = true;
+
+                        // Switch to the pane of the Output window
+                        outputPane.Activate();
+
+                        // Show the output window
+                        if (shell != null)
+                        {
+                            Guid commandSet = VSConstants.GUID_VSStandardCommandSet97;
+                            object inputVariant = null;
+                            shell.PostExecCommand(commandSet, (uint)VSConstants.VSStd97CmdID.OutputWindow, 0, ref inputVariant);
+                        }
+                    }
+
+                    // Output the text
+                    outputPane.OutputString(message);
+                }
+                catch (Exception)
+                {
```

Comment:
why blanket catch with no action? are you expecting a throw? Maybe a debug.assert here to call out why its throwing?

This seems different than the experience above where you are throwing NRE.